Add options for RATO REST API capacity

the inbo/rato-occurrences repo is failing on a reset by peer error:


```bash
4/4 [Fetch raw data via API call using ratatouille package]
output file: /tmp/Rtmpp1yjkd/file9205672df559.R
Fetching ■                                  0% | ETA:  1h
Fetching ■                                  1% | ETA:  4m
Fetching ■■■                                7% | ETA:  1m
Fetching ■■■■■                             14% | ETA:  1m
Fetching ■■■■■■■                           20% | ETA: 44s
Fetching ■■■■■■■■■                         26% | ETA: 40s
Fetching ■■■■■■■■■■■                       32% | ETA: 36s
Fetching ■■■■■■■■■■■                       34% | ETA: 35s
Error in `httr2::req_perform_parallel()` at ratatouille/R/get_objects.R:75:3:
! Failed to perform HTTP request.
Caused by error:
! Failure when receiving data from the peer [gis.oost-vlaanderen.be]:
Recv failure: Connection reset by peer
Backtrace:
     ▆
  1. ├─base::source("inst/source/run_fetch_data.R")
  2. │ ├─base::withVisible(eval(ei, envir))
  3. │ └─base::eval(ei, envir)
  4. │   └─base::eval(ei, envir)
  5. ├─base::source(tempR)
  6. │ ├─base::withVisible(eval(ei, envir))
  7. │ └─base::eval(ei, envir)
  8. │   └─base::eval(ei, envir)
  9. └─ratatouille::ratatouille(source = "rato")
 10.   └─ratatouille::get_objects(...) at ratatouille/R/ratatouille.R:15:3
 11.     ├─purrr::map(...) at ratatouille/R/get_objects.R:75:3
 12.     │ └─purrr:::map_("list", .x, .f, ..., .progress = .progress)
 13.     │   └─purrr:::vctrs_vec_compat(.x, .purrr_user_env)
 14.     └─httr2::req_perform_parallel(objects_requests, progress = "Fetching") at ratatouille/R/get_objects.R:75:3
 15.       └─rlang::cnd_signal(queue$resps[[i]])
Execution halted
Error: Process completed with exit code 1.
```
It seems to fail right as the bucket is filling up, maybe setting a lower capacity might help? 

It would be convenient if this is possible via an environmental variable.

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,6 +1,8 @@
 # ratatouille (development version)
 - `magrittr` is no longer a direct dependency, the package now uses base pipes 
 `|>` internally. (#41)
+- The RATO REST API capacity can now be set with an `options()` or by setting 
+the environmental variable `RATO_API_CAPACITY`. (#43)
 # ratatouille 1.0.0
 - Add `ratatouille()`, a function to fetch all available invasive species data 
 from RATO in a single function call. (#36)

--- a/R/get_objects.R
+++ b/R/get_objects.R
@@ -67,7 +67,10 @@ get_objects <- function(object_ids = list_object_ids(),
     }) |>
     # Set capacity of API: handle capacity outstanding requests, then wait for
     # requests to finish
-    purrr::map(~ httr2::req_throttle(.x, capacity = 1000)) |>
+    purrr::map( ~ httr2::req_throttle(.x,
+                                      capacity = getOption(
+                                        "ratatouille.RATO_API_CAPACITY")
+                                      )) |>
     # max_tries needs to be 2 for req_parallel
     purrr::map(~ httr2::req_retry(.x, max_tries = 3))
 

--- a/R/utils.R
+++ b/R/utils.R
@@ -36,7 +36,8 @@ as_datetime <- function(miliseconds, origin = "1970-01-01", ...) {
   op <- options()
   op.ratatouille <- list(
     ratatouille.rato_expires_minutes = 5,
-    ratatouille.cache_max_age_secs = 150
+    ratatouille.cache_max_age_secs = 150,
+    ratatouille.RATO_API_CAPACITY = 150
   )
   toset <- !(names(op.ratatouille) %in% names(op))
   if (any(toset)) options(op.ratatouille[toset])

